src/rollup/types.d.ts

```diff
@@ -25,15 +25,17 @@ export interface RollupError {
 	pluginCode?: string;
 }
 
-export interface RawSourceMap {
-	version: string;
-	sources: string[];
-	names: string[];
-	sourceRoot?: string;
-	sourcesContent?: string[];
-	mappings: string;
-	file: string;
-}
+export type RawSourceMap =
+	| { mappings: '' }
+	| {
+			version: string;
+			sources: string[];
+			names: string[];
```

Comment:
If we instead make the above all optional that might work just as well.

Reply:
I'm afraid not.

If either `source` or `names` is `undefined` when `mappings` is not an empty string, an error is raised [here](https://github.com/rollup/rollup/blob/8ba819f23d46a5a25015b2225aef39be67d0c06a/src/utils/getOriginalLocation.ts#L21-L22).